inc/sitemaps/class-sitemaps-cache.php

```diff
@@ -54,17 +54,59 @@ public function get_sitemap( $type, $page ) {
 	}
 
 	/**
+	 * Get the sitemap that is cached
+	 *
+	 * @param string $type Sitemap type.
+	 * @param int    $page Page number to retrieve.
+	 *
+	 * @return null|WPSEO_Sitemap_Cache_Data Null on no cache found otherwise object containing sitemap and meta data.
+	 */
+	public function get_sitemap_data( $type, $page ) {
+
+		$sitemap = $this->get_sitemap( $type, $page );
+
+		if ( empty( $sitemap ) ) {
+			return null;
+		}
+
+		// This should happend automatically, but it doesn't.
+		if ( 0 === strpos( $sitemap, 'C:24:"WPSEO_Sitemap_Cache_Data"' ) ) {
+			$sitemap = unserialize( $sitemap );
+		}
+
+		// What we expect it to be if it is set.
+		if ( $sitemap instanceof WPSEO_Sitemap_Cache_Data_Interface ) {
+			return $sitemap;
+		}
+
+		// Convert old type to new type, this will invalidate soon enough.
+		if ( is_string( $sitemap ) ) {
+			$sitemap_data = new WPSEO_Sitemap_Cache_Data();
+			$sitemap_data->set_sitemap( $sitemap );
+
+			return $sitemap_data;
+		}
+
+		return null;
+	}
+
+	/**
 	 * Store the sitemap page from cache.
 	 *
 	 * @param string $type    Sitemap type.
 	 * @param int    $page    Page number to store.
 	 * @param string $sitemap Sitemap body to store.
+	 * @param bool   $usable  Is this a valid sitemap or a cache of an invalid sitemap.
 	 *
 	 * @return bool
 	 */
-	public function store_sitemap( $type, $page, $sitemap ) {
+	public function store_sitemap( $type, $page, $sitemap, $usable = true ) {
+
+		$sitemap_data = new WPSEO_Sitemap_Cache_Data();
+		$sitemap_data->set_sitemap( $sitemap );
+		$sitemap_data->is_usable( $usable );
```

Comment:
If I understand correctly, you're not doing anything with the return value here. So I'm unsure why you are calling `set_sitemap` and `is_usable`. Could you please elaborate?